test/junit/scala/collection/immutable/StringLikeTest.scala

```diff
@@ -0,0 +1,28 @@
+package scala.collection.immutable
+
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+import scala.tools.testing.AssertUtil
+import scala.util.Random
+
+/* Test for SI-8988 */
+@RunWith(classOf[JUnit4])
+class StringLikeTest {
+  @Test
+  def testStringSplitWithChar: Unit = {
+    val chars = (0 to 255).map(_.toChar)
+    def randString = Random.nextString(30)
+
+    for (c <- chars) {
+      val s = randString
+      val jString = new java.lang.String(s)
+
+      // make sure we can match a literal character done by Java's split
+      val jSplit = jString.split("\\Q" + c.toString + "\\E")
+      val sSplit = s.split(c)
```

Comment:
This doesn't even use the new code, because `split` with a char argument is a method on `java.lang.String`, and `s` is just a `java.lang.String`.  You need to have something else `StringLike`--`StringBuilder` for instance.